pkg/kubelet/dockershim/docker_container.go

```diff
@@ -416,9 +417,16 @@ func (ds *dockerService) ContainerStatus(_ context.Context, req *runtimeapi.Cont
 	}
 
 	labels, annotations := extractLabels(r.Config.Labels)
+	imageInSpec := annotations[types.KubernetesContainerImageLabel]
```

Comment:
It is a common problem for all the runtime. Would it be better to put the whole restore image name to `toKubeContainerStatus`? Then other kube runtimes(e.g. remote runtime) could also benefit from it. https://github.com/kubernetes/kubernetes/blob/c46c1c043ee8de4c021c03ddb0fb157d1c1ba3f8/pkg/kubelet/kuberuntime/kuberuntime_container.go#L484-L498